include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -186,297 +186,242 @@ class __kernel_name_base
     {
 #if 0 //_ONEDPL_KERNEL_BUNDLE_PRESENT
         auto __kernel_bundle = sycl::get_kernel_bundle<sycl::bundle_state::executable>(__exec.queue().get_context());
-        return __kernel_bundle.get_kernel(sycl::get_kernel_id<_DerivedKernelName>());
+        return __kernel_bundle.get_kernel(sycl::get_kernel_id<_DerivedKernelName...>());
 #else
         sycl::program __program(__exec.queue().get_context());
 
-        __program.build_with_kernel_type<_DerivedKernelName>();
-        return __program.get_kernel<_DerivedKernelName>();
+        __program.build_with_kernel_type<_DerivedKernelName...>();
+        return __program.get_kernel<_DerivedKernelName...>();
 #endif
     }
 };
 
-template <typename... _Name>
-class __parallel_for_kernel : public __kernel_name_base<__parallel_for_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_reduce_kernel : public __kernel_name_base<__parallel_reduce_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_scan_local_kernel : public __kernel_name_base<__parallel_scan_local_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_scan_global_kernel : public __kernel_name_base<__parallel_scan_global_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_scan_propagate_kernel : public __kernel_name_base<__parallel_scan_propagate_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_find_or_kernel : public __kernel_name_base<__parallel_find_or_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_merge_kernel : public __kernel_name_base<__parallel_merge_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_sort_leaf_kernel : public __kernel_name_base<__parallel_sort_leaf_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_sort_global_kernel : public __kernel_name_base<__parallel_sort_global_kernel<_Name...>>
-{
-};
-template <typename... _Name>
-class __parallel_sort_copy_back_kernel : public __kernel_name_base<__parallel_sort_copy_back_kernel<_Name...>>
-{
-};
 
-//------------------------------------------------------------------------
-// parallel_for - async pattern
-//------------------------------------------------------------------------
 
-//General version of parallel_for, one additional parameter - __count of iterations of loop __cgh.parallel_for,
```

Comment:
This 2-lines comment is important for "parallel_for" pattern.

Reply:
Done